pkg/kubectl/cmd/apply_set_last_applied.go

```diff
@@ -185,7 +186,7 @@ func (o *SetLastAppliedOptions) RunSetLastApplied(f cmdutil.Factory, cmd *cobra.
 				return err
 			}
 			helper := resource.NewHelper(client, mapping)
-			patchedObj, err := helper.Patch(o.Namespace, info.Name, types.MergePatchType, patch)
+			patchedObj, err := helper.Patch(o.Namespace, info.Name, types.StrategicMergePatchType, patch)
```

Comment:
merge patch is cheaper to compute and apply, and we don't need the power of strategic merge patch here